server/src/models/services/populate.js

```diff
@@ -166,11 +181,110 @@ export default async function ({ models }) {
       ),
     })
   );
+  const group_by_program_id = (result, service) => {
+    _.forEach(service.program_ids, (program_id) => {
+      result[program_id] = result[program_id]
+        ? _.concat(result[program_id], service)
+        : [service];
+    });
+    return result;
+  };
+  const service_types_lookup = _.chain(
+    get_standard_csv_file_rows("service_types_lookup.csv")
+  )
+    .map(({ code, en, fr }) => [code, { en, fr }])
+    .fromPairs()
+    .value();
+  const gov_type_summary = _.chain(service_rows)
+    .flatMap("service_type_code")
+    .countBy()
+    .map((value, type_code) => ({
+      id: `${type_code}_${value}`,
+      subject_id: "gov",
+      label_en: service_types_lookup[type_code].en,
+      label_fr: service_types_lookup[type_code].fr,
+      value,
+    }))
+    .value();
+  const dept_type_summary = _.chain(service_rows)
+    .groupBy("org_id")
+    .flatMap((services, org_id) =>
+      _.chain(services)
+        .flatMap("service_type_code")
+        .countBy()
+        .map((value, type_code) => ({
+          id: `${org_id}_${type_code}_${value}`,
+          subject_id: org_id,
+          label_en: service_types_lookup[type_code].en,
+          label_fr: service_types_lookup[type_code].fr,
+          value,
+        }))
+        .value()
+    )
+    .value();
+  const program_type_summary = _.chain(service_rows)
+    .reduce(group_by_program_id, {})
+    .flatMap((services, program_id) =>
+      _.chain(services)
+        .flatMap("service_type_code")
+        .countBy()
+        .map((value, type_code) => ({
+          id: `${program_id}_${type_code}_${value}`,
+          subject_id: program_id,
+          label_en: service_types_lookup[type_code].en,
+          label_fr: service_types_lookup[type_code].fr,
+          value,
+        }))
+        .value()
+    )
+    .value();
+  const get_current_status_count = (services, key, value) =>
+    _.countBy(services, `${key}_status`)[value] || 0;
+  const populate_digital_summary_key = (services, subject_id, level, key) => ({
+    id: `${level}_${subject_id}_${key}`,
+    key_desc: `${key}_desc`,
+    key,
+    subject_id,
+    can_online: get_current_status_count(services, key, true),
+    cannot_online: get_current_status_count(services, key, false),
+    not_applicable: get_current_status_count(services, key, null),
```

Comment:
(1) Potential red flags: these `key_desc`, `can_online`, `cannot_online`, `not_applicable` are keys that is expected and assumed to be translated over in client side. Should this be taken care of on the server side?

Reply:
I think doing that on the client makes more sense